include/ignition/math/Vector3.hh

```diff
@@ -264,6 +264,14 @@ namespace ignition
         return n.Normalize();
       }
 
+      /// \brief Get Projection of another onto this vector
+      /// \param[in] _v the vector
+      /// \return the projection
+      public: T Project(const Vector3<T> &_v) const
+      {
+        return this->Dot(_v) / this->Length();
```

Comment:
Nope. Not necessarily https://en.wikipedia.org/wiki/Vector_projection .

Reply:
What part says that the projection could be longer? The scalar projection is the length of the vector multiplied by the cosine of the angle between the 2 vectors, which is always <= 1, right?

Reply:
I should have been clearer in naming. Its not a Scalar projection.its the length of the Vector projection. A search of the codebase also shows I no longer use this function. Perhaps I should remove it.

Reply:
I'm ok removing it if it's not needed now, it can always be added later :+1:

Reply:
Apologies, this class is used in calculating the box volume. And it does need to be the length of the VectorProjection.

Reply:
I think I got confused with the test expectations because I was thinking that on that 2nd expectation, `v2` was projected onto `v3`, so it was weird that the projection had length 3 while the original vector had length 1.

https://github.com/ignitionrobotics/ign-math/blob/cb912344a4974ccfc7e4ae2938518dcc3cd63c06/src/Vector3_TEST.cc#L441-L447

The order of operations also makes sense for me now, sorry for the noise.

> Its not a Scalar projection.its the length of the Vector projection

I _think_ those should be the same, according to Wikipedia:

"_The vector projection of a on b is a vector whose magnitude is the scalar projection of a on b with the same direction as b."_

Reply:
I'm concerned that this will result in a divide by zero if `this` is a Zero vector

what should this method return in that case?

Reply:
Hmm technically I don't think Vector Projections are defined for zero length vectors. I could return zero but I'm not sure thats the best way to go around it.

Reply:
considering where you are using this method; do you currently check the length of the vector before calling this method?

Reply:
Its now used in more than one function. I've added a doc-string warning that ts can return nans.

Reply:
>> Another option is not adding this function to `Vector3` for now and just calculating it directly within that function for now.

> Its now used in more than one function. I've added a doc-string warning that ts can return nans.

I see that it is used in multiple places, but I don't see any logic that checks for NaNs. Since this function is currently only one line, I would prefer to redact this API and duplicate this one line of code with context-specific error-checking in the two functions where it is currently used

Reply:
Removed the function and inlined it.